### PR TITLE
Various Improvements

### LINKED ISSUE #4: Avoid duplicate commenting
Consider:
- someone is invited
- chooses not to accept
- sends a second PR

GitHub's API won't let us spam them with duplicate invitations, but we shouldn't send a second comment inviting them.

This issue has only happened once in the past year of Moya, but it's still worth addressing.

### PR DESCRIPTION
**Pull request by ashfurrow**
This moves to a new method of checking for team membership that's more specific, so we don't need the `ORG_NAME` env variable anymore. Also avoids unpacking several dictionary lookups unless we're sure they're not nil (i.e. `push['pull_request']['merged']` would fail if `['pull_request'] lookup returns`nil`).

Also fixes #4, which would comment on PRs opened by users who had already been invited but chosen not to accept.

Edit: also adds Rubocop.

### CODE REVIEW

File: spec/server_spec.rb [hunk 1]
```diff
@@ -19,34 +19,34 @@
...
+    allow(ping_checker).to receive(:ping?).and_return(false)
+    allow(api).to receive(:handle_push).and_return('msg' => 'Everything is fine.')
+
     post '/payload', '{}'
 
     expect(last_response).to be_ok
-    expect(last_response.status) == 200
-    expect(JSON.parse(last_response.body)['error']) == 'Everything is fine.'
+    expect(last_response.status).to eq(200)
+    expect(JSON.parse(last_response.body)['msg']).to eq('Everything is fine.')
```
Comment (ashfurrow) [inline spec/server_spec.rb:39]: This used to say `error` but was changed to `msg` after Rubocop pointed out a problem using `==` (where the lhs is `nil`). Neat.

File: Gemfile [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
 
 gem 'sinatra'
 gem 'json'
 gem 'octokit'
 
-group :test do 
+group :development do
+  gem 'rubocop'
+  gem 'guard'
+  gem 'guard-rubocop'
```
Comment (orta) [inline Gemfile:11]: in theory, you could just have `guard-rubocop` - both the other deps would come from that

File: Guardfile [hunk 3]
```diff
@@ -12,3 +12,8 @@ guard :rspec, cmd: "bundle exec rspec" do
   ruby = dsl.ruby
   dsl.watch_spec_files_for(ruby.lib_files)
 end
+
+guard :rubocop do
+  watch(/.+\.rb$/)
```
Comment (orta) [inline Guardfile:17]: aye, I need to do this on Danger

File: lib/api.rb [hunk 4]
```diff
@@ -2,36 +2,44 @@
...
+  rescue Octokit::NotFound
+    return true
   end
 
-  def is_merged?(push)
-    push['action'] == 'closed' && push['pull_request']['merged'] == true
+  def invite_and_comment(team_id, username, pr_number, repo_name)
+    @github_client.add_team_membership(team_id, username)
+    @github_client.add_comment(repo_name, pr_number, ENV['INVITATION_MESSAGE'])
   end
```
Comment (orta) [inline lib/api.rb:44]: so, let's talk about `@github` - I'll talk in Obj-C for metaphor.

``` objc
@interface API
@property (nonatomic, strong) Client *github;
@end
```

You effectively have this, and throughout the source code, you're always accessing `_github` the raw instance variable, this aint common parlance for Ruby. You have an implicit set/getter via the  `attr_accessor :github_client` so use them. Free yourself of the `@`. You can make them `github` instead as it resolves to the getter method.

Note, you have to set it either via `@github` or `self.github`.

**Comment by orta** [comment]
Mergable depending on feels for your usage on instance vars

File: lib/api.rb (hunk 4 above)
Comment (ashfurrow) [inline lib/api.rb:44]: Ah, didn't know that. Thanks!!

**Comment by orta** [comment]
👍

**Pull request merged**